storage/src/vespa/storage/bucketdb/storagebucketdbinitializer.cpp

```diff
@@ -429,7 +430,8 @@ StorageBucketDBInitializer::registerBucket(const document::Bucket &bucket,
             bucketId.toString().c_str(), int(partition));
         entry->disk = partition;
         entry.write();
-        uint16_t disk(_system._distribution.getIdealDisk(
+        auto distribution(contentBucketSpace.getDistribution());
+        uint16_t disk(distribution->getIdealDisk(
```

Comment:
Just a note that this will incur an additional lock + atomic inc/dec per bucket since we're no longer caching the distribution `shared_ptr`. Probably not an issue since this code path is only triggered at startup and the mutex-covered region is minimal.